### PR TITLE
Adding K-NN in 3.0.0 manifest and geospatial in 2.4 and 3.0 manifests.

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Adding K-NN in 3.0.0 manifest and geospatial in 2.4 and 3.0 manifests.

Signed-off-by: Navneet Verma <navneev@amazon.com>

### Issues Resolved
https://github.com/opensearch-project/geospatial/issues/94

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by heemin32 — APPROVED** [review]

File: manifests/2.4.0/opensearch-2.4.0.yml [hunk 1]
```diff
@@ -83,3 +83,11 @@ components:
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
+  - name: geospatial
+    repository: https://github.com/opensearch-project/geospatial.git
+    ref: '2.x'
+    platforms:
+      - linux
```
Comment (zelinh) [inline manifests/2.4.0/opensearch-2.4.0.yml:90]: I think we would need to add `windows` platform for 2.4 manifest as windows distribution is planned to released in this version. @peterzhuamazon

Comment (peterzhuamazon) [reply manifests/2.4.0/opensearch-2.4.0.yml:90]: Yes.

Comment (peterzhuamazon) [reply manifests/2.4.0/opensearch-2.4.0.yml:90]: Also the groovy failure might not relate to this tho.

File: manifests/3.0.0/opensearch-3.0.0.yml [hunk 2]
```diff
@@ -32,3 +32,19 @@ components:
     checks:
       - gradle:properties:version
       - gradle:dependencies:opensearch.version
+  - name: geospatial
+    repository: https://github.com/opensearch-project/geospatial.git
+    ref: main
+    platforms:
+      - linux
```
Comment (peterzhuamazon) [inline manifests/3.0.0/opensearch-3.0.0.yml:39]: Please add windows here as well.

Comment (peterzhuamazon) [reply manifests/3.0.0/opensearch-3.0.0.yml:39]: Only add for geospatial leave knn linux only.

**Comment by jordarlu** [comment]
@navneet1v , could you rebase your repo with opensearch-build@main and rerun the validation?  it seems like your repo is out dated with the opensearch-build@main; hence, groovy test was fail.  Thanks.

**Comment by navneet1v** [comment]
I can see that @naveentatikonda has raised another PR for the same and it got merged. https://github.com/opensearch-project/opensearch-build/commit/6b71a5ff3e2065fdc527aa31fcf5d490db94503e Closing this PR.

**Pull request closed without merging**